### ISSUE TITLE
Question about text classification loss and image regression loss

### ISSUE THREAD

**Issue by xmy0916** [baaivision/Emu#56]
![image](https://github.com/baaivision/Emu/assets/43675899/f609d352-f087-4a85-8264-39a8863c08bf)
Thanks for your great job! 
I don’t understand the structure of the paper a bit. Why is there a loss of text classification here? And what is the specific loss of this image regression?

**Comment by xmy0916** [comment]
In addition, I see that the paper sets different learning rates for the LLM, vision encoder and projector layers in the pre-training stage, so all model parameters need to be trained in the pre-training stage, right? If so, I’m curious about how 80GB of video memory can be used for training. Do you have any training techniques you can share? Thanks

**Comment by ryanzhangfan** [comment]
### Loss
- **Emu2** is designed to possess both strong visual understanding and visual generation capabilities. For visual understanding, **Emu2** is desired to answer the questions based on image and text prompt inputs, which requires a strong text generation ability. So besides image regression loss, we also need a text classification loss here.

- We use MSE loss for image token regression.

### About training
In fact we use multiple 80GB GPUs to train **Emu2**-series(from tens to hundreds depending on different training stages), adopting the DeepSpeed framework..

**Comment by xmy0916** [comment]
So the loss of text classification mentioned here is the loss of text generation, right?

**Comment by ryanzhangfan** [comment]
> So the loss of text classification mentioned here is the loss of text generation, right?

Yes.

**Comment by xmy0916** [comment]
> > So the loss of text classification mentioned here is the loss of text generation, right?
> 
> Yes.

Thanks
